cvat-ui/src/components/annotation-page/standard-workspace/controls-side-bar/draw-shape-popover-content.tsx

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+
+import {
+    Row,
+    Col,
+    Select,
+    Button,
+    InputNumber,
+} from 'antd';
+
+import Text from 'antd/lib/typography/Text';
+
+import {
+    ShapeType,
+    ObjectType,
+} from 'reducers/interfaces';
+
+import {
+    Canvas,
+} from 'cvat-canvas';
+
+interface Props {
+    canvasInstance: Canvas;
+    shapeType: ShapeType;
+    labels: {
+        [index: number]: string;
+    };
+    onDrawStart(
+        shapeType: ShapeType,
+        labelID: number,
+        objectType: ObjectType,
+        points?: number,
+    ): void;
+}
+
+interface State {
+    numberOfPoints?: number;
+    selectedLabeID: number;
+}
+
+export default class DrawShapePopoverContent extends React.PureComponent<Props, State> {
+    constructor(props: Props) {
+        super(props);
+        const defaultLabelID = +Object.keys(props.labels)[0];
+        this.state = {
+            selectedLabeID: defaultLabelID,
+        };
+    }
+
+    public render(): JSX.Element {
+        const {
+            numberOfPoints,
+            selectedLabeID,
+        } = this.state;
+
+        const {
+            shapeType,
+            labels,
+            onDrawStart,
+            canvasInstance,
+        } = this.props;
+
+        let minimumPoints = 0;
```

Comment:
Maybe me move this condition to function out of render method? Like so:
```javascript
public getMinimumPoints(shapeType): number {
        let minimumPoints = 0;
        if (shapeType === ShapeType.POINTS) {
            minimumPoints = 1;
        } else if (shapeType === ShapeType.POLYLINE) {
            minimumPoints = 2;
        } else if (shapeType === ShapeType. POLYGON) {
            minimumPoints = 3;
        }
        return minimumPoints;
}

```

Reply:
Yes